tests/test_bqpipeline.py

```diff
@@ -63,24 +77,116 @@ def test_create_job_config_destination(self):
             self.assertEqual(cfg.destination.project, 'testproject')
 
     def test_create_job_config_flags(self):
-        bq = BQPipeline(
-            job_name='testjob', default_project='testproject', default_dataset='testdataset')
+        bq = bqpipeline.BQPipeline(
+            job_name='testjob', default_project='testproject',
+            default_dataset='testdataset')
         cfg = bq.create_job_config(batch=False, create=False, overwrite=False)
         self.assertEqual(cfg.priority, bigquery.QueryPriority.INTERACTIVE)
         self.assertEqual(cfg.create_disposition,
                          bigquery.job.CreateDisposition.CREATE_NEVER)
         self.assertEqual(cfg.write_disposition,
                          bigquery.job.WriteDisposition.WRITE_EMPTY)
 
+    def test_scalar_parameters(self):
+        result = bqpipeline.set_scalar_parameter('test', 'abc', typ='str')
+        self.assertEqual(result, bigquery.ScalarQueryParameter('test',
+                                                               'STRING', 'abc'))
+
+        result = bqpipeline.set_scalar_parameter('test', 1, typ='int')
+        self.assertEqual(result, bigquery.ScalarQueryParameter('test',
+                                                               'INT64', 1))
+
+        dt = datetime.datetime.now()
+        result = bqpipeline.set_scalar_parameter('test', dt, typ='datetime')
+        self.assertEqual(result, bigquery.ScalarQueryParameter('test',
+                                                               'TIMESTAMP', dt))
+
+        b = bytes(123)
+        result = bqpipeline.set_scalar_parameter('test', b, typ='bytes')
+        self.assertEqual(result, bigquery.ScalarQueryParameter('test',
+                                                               'BYTES', b))
+
+        with self.assertRaises(KeyError):
+            bqpipeline.set_scalar_parameter('test', 1, typ='unsupported')
+
+    def test_array_parameters(self):
+        result = bqpipeline.set_list_parameter('test', ['abc'], typ='list')
+        self.assertEqual(result, bigquery.ArrayQueryParameter('test', 'STRING',
+                                                              ['abc']))
+
+        result = bqpipeline.set_list_parameter('test', [1, 2], typ='list')
+        self.assertEqual(result, bigquery.ArrayQueryParameter('test', 'INT64',
+                                                              [1, 2]))
+
+        with self.assertRaises(IndexError):
```

Comment:
See suggested refactoring in function def. This should raise `ValueError`
Also add test for a list with different types ie `[1, 'two', False]`.

Reply:
This is done in the test_query_params unit test. I've added additional unit tests with more scenarios for other types.